netbox/netbox/models/__init__.py

```diff
@@ -58,6 +59,32 @@ class NetBoxModel(CloningMixin, NetBoxFeatureSet, models.Model):
     class Meta:
         abstract = True
 
+    def clean(self):
+        """
+        Validate the model for GenericForeignKey fields to ensure that the content type and object ID exist.
+        """
+        super().clean()
+
+        for field in self._meta.get_fields():
+            if isinstance(field, GenericForeignKey):
+                if getattr(self, field.ct_field) is None and getattr(self, field.fk_field) is not None:
```

Comment:
Instead of calling `getattr()` for either field each time, we can just call it twice to define `ct_field` and `fk_field` at the beginning and then reference those values.